rest/route/host_agent.go

```diff
@@ -483,6 +485,39 @@ func assignNextAvailableTask(ctx context.Context, env evergreen.Environment, tas
 			return nil, true, nil
 		}
 
+		if currentHost.Distro.SingleTaskDistro {
+			// If next task exists and the distro is a single task distro, check if the task is allowed on the distro.
+			allowedTasks, err := validator.GetAllowedSingleTaskDistroTasksForProject(ctx, nextTask.Project)
+			if err != nil {
+				grip.Alert(message.WrapError(err, errMsg))
+				return nil, false, errors.Wrapf(err, "could not find allowed single task disto tasks for project '%s'", nextTask.Project)
+			}
+			matched := false
+			for _, allowedTask := range allowedTasks {
+				matched, err = regexp.MatchString(allowedTask, nextTask.DisplayName)
+				if err != nil {
+					grip.Alert(message.WrapError(err, errMsg))
+					return nil, false, errors.Wrapf(err, "could not process regex '%s'", allowedTask)
+				}
+				if matched {
+					break
+				}
+			}
+			if !matched {
+				grip.Alert(message.Fields{
```

Comment:
If we just alert here, it'll still leave the task at the top of the queue unable to dispatch, so this will keep alerting unless someone manually intervenes to unschedule it. Can we instead consider unscheduling or system-failing the task? We could call [`MarkEnd`](https://github.com/evergreen-ci/evergreen/blob/2305f76b19d6499d1430ce03ad5a4837e90528b3/model/task_lifecycle.go#L770) to kick the task out of the queue.

Reply:
done